### PR TITLE
D8AGE-581: Enhance the Request Updater test to check optional values.

### PR DESCRIPTION
**Pull request by intelektron**
## D8AGE-581

The follow-up of https://github.com/openeuropa/oe_translation/pull/322

### CODE REVIEW

File: modules/oe_translation_cdt/tests/src/Kernel/TranslationRequestUpdaterTest.php [hunk 1]
```diff
@@ -221,6 +228,7 @@ protected function createTranslationRequest(): TranslationRequestCdtInterface {
    *   The expected log messages. There may be wildcards "*" inside.
    */
   protected function assertTranslationRequestLog(TranslationRequestCdtInterface $request, array $logs): void {
+    $this->assertEquals(count($logs), count($request->getLogMessages()), 'The log size doesn\'t match.');
```
Comment (drishu) [inline modules/oe_translation_cdt/tests/src/Kernel/TranslationRequestUpdaterTest.php]: assertSameSize

**Review by drishu — CHANGES_REQUESTED** [review]
We have two createTranslationRequest we should sync their usage into one function and put it into a trait

File: modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ */
+trait CdtTranslationTestTrait {
+
+  /**
+   * Gets the common translation request parameters.
+   *
+   * @return array
+   *   The common parameters of CDT translation request.
+   */
+  private function getCommonTranslationRequestData() {
```
Comment (drishu) [inline modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php]: add return type hinting, and I think we can make it static

File: modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_translation_cdt;
```
Comment (drishu) [inline modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php]: this should go into a "Traits" folder

File: modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+   *   The data to create the translation request.
+   * @param \Drupal\Core\Entity\ContentEntityInterface|null $entity
+   *   The content entity to attach to the translation request.
+   * @param array $languages
+   *   The target languages.
+   *
+   * @return \Drupal\oe_translation_cdt\TranslationRequestCdtInterface
+   *   The translation request.
+   */
+  private function createTranslationRequest(array $data, ?ContentEntityInterface $entity, array $languages): TranslationRequestCdtInterface {
```
Comment (drishu) [inline modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php]: optional parameters should go last

**Review by drishu — CHANGES_REQUESTED** [review]

File: modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php (hunk 4 above)
Comment (intelektron) [reply modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php]: This one is only nullable

Comment (drishu) [reply modules/oe_translation_cdt/tests/src/CdtTranslationTestTrait.php]: nullable and default value both make the parameter optional, but not worth blocking for, so I'll resolve this

**Review by drishu — APPROVED** [review]

**Pull request merged**